java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -123,6 +132,42 @@ protected String handleStringResponse(Response response) {
                         + " expected String");
     }
 
+    /**
+     * Extracts the response value from the Redis response and either throws an exception or returns
+     * the value as an <code>Long</code>.
+     *
+     * @param response Redis protobuf message
+     * @return Response as an <code>Long</code>
+     */
+    protected Long handleLongResponse(Response response) {
+        Object value = handleObjectResponse(response);
+        if (value instanceof Long) {
+            return (Long) value;
+        }
+        throw new RedisException(
+                "Unexpected return type from Redis: got "
```

Comment:
We should consider nulls here:
https://github.com/aws/glide-for-redis/blob/42b0498da2e1ef89947cf529d4e56b0893b91904/java/client/src/main/java/glide/api/BaseClient.java#L148